teps/0060-remote-resource-resolution.md

```diff
@@ -242,207 +255,562 @@ users can leverage those shared pipelines.
+
+The "bundle" resolver will support looking up resources from Tekton
+Bundles. This will mirror the existing support in Tekton Pipelines. In
+addition it will verify a bundle before returning it if configured to do
+so. The precise approach to verifying a bundle will be based on the
+decisions made in
+[TEP-0091](https://github.com/tektoncd/community/pull/537).
+
+The supported paramaters for a "bundle" `ResourceRequest` will be:
+
+```yaml
+image_url: gcr.io/tekton-releases/catalog/upstream/golang-fuzz:0.1
+name: golang-fuzz
+signer: cosign # TBD, see TEP-0091
+```
+
+When the bundle is successfully resolved the `ResourceRequest` will be
+updated with both the resource contents and an annotation with the
+digest of the fetched image.
+
+#### Git Resolver
+
+The "git" resolver will support fetching resources from git
+repositories. This is new functionality that is not supported by Tekton
+Pipelines currently. In addition to basic support for fetching files
+from git this resolver will also need to support quite rich
+configuration (which can be developed over time):
+
+- Allow-lists for specific repo urls, branches and commits.
+- Settings for limiting access to certain repos by namespace.
+- Timeout settings for git operations.
+- Proxy settings (equivalent to `HTTP_PROXY`, `HTTPS_PROXY` environment
+  variables).
+- Path filtering such that only specific directories and file paths
+  within a repo can be sourced.
+- Verification of fetched files, following any approach decided in
+  [TEP-0091](https://github.com/tektoncd/community/pull/537).
+
+Since clones of large repos can be slow, and not all providers support
+features like sparse checkout, the "git" resolver will implement caching
+as a high priority.
+
+The parameters for a "git" `ResourceRequest` will initially look as
+follows:
+
+```yaml
+repository_url: https://github.com/tektoncd/catalog.git
+commit: 2c9b093e94f30f588dc798cc56a2559d4da9d573
+branch: main # either commit or branch may be specified, but not both
+path: /task/golang-fuzz/0.1/golang-fuzz.yaml
+```
+
+The git resolver will need to gracefully handle concurrent requests for
+the same resource and will also need to be able to cancel in-flight
+operations if a `ResourceRequest` is failed or deleted.
+
+When the git resource is successfully resolved the `ResourceRequest`
+will be updated with both the resource contents as well as an annotation
+with the fetched commit SHA.
+
+### Pipelines' "Out of the Box" experience
+
+One of the goals for this feature is to reach a point of maturity that
+Tekton Pipelines could consider replacing its baked-in `taskRef` and
+`pipelineRef` support with resolvers from the Tekton Resolution project.
+
+At that point Pipelines' maintainers would need to decide how best to
+provide default resolvers "out of the box". One possible approach would be
+to deploy the `ResourceRequest` lifecycle reconciler and a set of resolvers
+as part of Pipelines' `release.yaml`.
 
 ## Test Plan
```

Comment:
Until now we have not provided users with a recommended way to manage their `Tasks`. 
I think it would be great to provide some "typical task management" examples, one per task provider, and focus some of our tests around them. We could setup different part of `dogfooding` with different approaches.

Reply:
Awesome, I've added an additional paragraph describing this to the Dogfooding portion of the Test Plan section:

```diff
+ Once Remote Resolution is supported in Dogfooding we can also implement
+ testing for recommended "Task Management" approaches. Documenting these
+ will also provide guidelines to help the community structure their own
+ repos and registries of `Pipelines` and `Tasks`.
```